bgp-monitor-action-pack/bgp-monitor/script.py

```diff
@@ -55,6 +56,31 @@ def IsStatsDiffExpected(prevStats, currentStats, expectedDiff: int):
         err = "Missing change control device" if device is None \
             else f"device {device} is missing 'id'"
         raise ActionFailed(err)
+
+    # This block checks for the check_established flag, defaults to False
+    # If set to True, this block will check that all BGP peers are "Established"
+    # If any BGP peers are not Established, the CC will abort/fail
+    if checkEstablished:
```

Comment:
Is there a case where we want to wait or retry if we find a peer isn't up?  Eg, should we run this in a loop for a configurable number of times before calling it a failure?

Reply:
I did think of that, in the case of the user in the field they did not see that being necessary. But I definitely agree that should be considered, basically like a "check_wait" for how long to wait for the state to correct to Established before aborting the CC.